llama-index-integrations/llms/llama-index-llms-maritalk/llama_index/llms/maritalk/base.py

```diff
@@ -120,108 +172,241 @@ def chat(self, messages: Sequence[ChatMessage], **kwargs: Any) -> ChatResponse:
+                        response_data = line.replace(b"data: ", b"").decode("utf-8")
+                        if response_data:
+                            parsed_data = json.loads(response_data)
+                            if "text" in parsed_data:
+                                content_delta = parsed_data["text"]
+                                content += content_delta
+                                yield ChatResponse(
+                                    message=ChatMessage(
+                                        role=MessageRole.ASSISTANT, content=content
+                                    ),
+                                    delta=content_delta,
+                                    raw=parsed_data,
+                                )
+            else:
+                raise MaritalkHTTPError(response)
 
-    def stream_chat(
-        self, messages: Sequence[ChatMessage], **kwargs: Any
-    ) -> ChatResponseGen:
-        raise NotImplementedError(
-            "Maritalk does not currently support streaming completion."
-        )
+        return gen()
 
+    @llm_completion_callback()
     def stream_complete(
         self, prompt: str, formatted: bool = False, **kwargs: Any
     ) -> CompletionResponseGen:
-        raise NotImplementedError(
-            "Maritalk does not currently support streaming completion."
-        )
-
-    @llm_chat_callback()
-    async def achat(
-        self, messages: Sequence[ChatMessage], **kwargs: Any
-    ) -> ChatResponse:
-        return self.chat(messages, **kwargs)
-
-    @llm_completion_callback()
-    async def acomplete(
-        self, prompt: str, formatted: bool = False, **kwargs: Any
-    ) -> CompletionResponse:
-        return self.complete(prompt, formatted, **kwargs)
+        stream_complete_fn = stream_chat_to_completion_decorator(self.stream_chat)
+        return stream_complete_fn(prompt, **kwargs)
 
     @llm_chat_callback()
     async def astream_chat(
         self, messages: Sequence[ChatMessage], **kwargs: Any
     ) -> ChatResponseAsyncGen:
-        raise NotImplementedError(
-            "Maritalk does not currently support streaming completion."
-        )
+        # Prepare the data payload for the Maritalk API
+        formatted_messages = []
+        for msg in messages:
+            if msg.role == MessageRole.SYSTEM:
+                # Add system message as a user message
+                formatted_messages.append({"role": "user", "content": msg.content})
+                # Follow it by an assistant message acknowledging it, to maintain conversation flow
+                formatted_messages.append({"role": "assistant", "content": "ok"})
+            else:
+                # Format user and assistant messages as before
+                formatted_messages.append(
+                    {
+                        "role": "user" if msg.role == MessageRole.USER else "assistant",
+                        "content": msg.content,
+                    }
+                )
```

Comment:
I think we don't need this if/else anymore as we now support SYSTEM messages in the API. Could you please take a look on the LangChain implementation to know how it is done? https://api.python.langchain.com/en/latest/_modules/langchain_community/chat_models/maritalk.html#ChatMaritalk